packages/flutter/lib/src/material/tabs.dart

```diff
@@ -397,29 +397,23 @@ class _IndicatorPainter extends CustomPainter {
     _needsPaint = false;
     _painter ??= indicator.createBoxPainter(markNeedsPaint);
 
-    if (controller.indexIsChanging) {
-      // The user tapped on a tab, the tab controller's animation is running.
-      final Rect targetRect = indicatorRect(size, controller.index);
-      _currentRect = Rect.lerp(targetRect, _currentRect ?? targetRect, _indexChangeProgress(controller));
-    } else {
-      // The user is dragging the TabBarView's PageView left or right.
-      final int currentIndex = controller.index;
-      final Rect previous = currentIndex > 0 ? indicatorRect(size, currentIndex - 1) : null;
-      final Rect middle = indicatorRect(size, currentIndex);
-      final Rect next = currentIndex < maxTabIndex ? indicatorRect(size, currentIndex + 1) : null;
-      final double index = controller.index.toDouble();
-      final double value = controller.animation.value;
-      if (value == index - 1.0)
-        _currentRect = previous ?? middle;
-      else if (value == index + 1.0)
-        _currentRect = next ?? middle;
-      else if (value == index)
-        _currentRect = middle;
-      else if (value < index)
-        _currentRect = previous == null ? middle : Rect.lerp(middle, previous, index - value);
-      else
-        _currentRect = next == null ? middle : Rect.lerp(middle, next, value - index);
-    }
+    // The user is dragging the TabBarView's PageView left or right.
+    final int currentIndex = controller.index;
+    final Rect previous = currentIndex > 0 ? indicatorRect(size, currentIndex - 1) : null;
+    final Rect middle = indicatorRect(size, currentIndex);
+    final Rect next = currentIndex < maxTabIndex ? indicatorRect(size, currentIndex + 1) : null;
+    final double index = controller.index.toDouble();
+    final double value = controller.animation.value;
+    if (value == index - 1.0)
+      _currentRect = previous ?? middle;
+    else if (value == index + 1.0)
+      _currentRect = next ?? middle;
+    else if (value == index)
+      _currentRect = middle;
+    else if (value < index)
+      _currentRect = previous == null ? middle : Rect.lerp(middle, previous, index - value);
```

Comment:
isn't this assume index - value is always <1.0 ? in case of user tapp on a tab, this value may be >1.0

Reply:
Yes you are right. `index - value` can be greater than `1.0`. I thought this was the solution for the case of tapping a tab that more than one tab away. If there are any reason it's not allowed, please let me know.

I added a gif that simulated the situation you were talking about as I understood it. If this is not the case of you intended, could you describe the case of tapping a tab that more than one tab away for reproduce?

![tab_indicator_test](https://user-images.githubusercontent.com/26322627/93846955-86a42680-fce0-11ea-8512-b13df20f2751.gif)

Reply:
When there are three tabs A, B, C, the image below is a scenario in which the user taps C and then taps B after the animation has progressed 25%.

![tab](https://user-images.githubusercontent.com/26322627/98914552-2a08ff00-250c-11eb-89ad-99b2240687fc.png)

Here you can see a case which the lerp value is greater than 1, and I think it can be relatively greater or less than 1 depending on which point the lerp is based on.

As [this document](https://api.flutter.dev/flutter/dart-ui/Rect/lerp.html) also says that a value greater than 1 or less than 0 is not an invalid value, I think it is just a difference in mathematical expressions and there is no problem.

Nevertheless, if the lerp value should be between 0 and 1, I think that the function should be newly implemented.

Reply:
This seems to not working correctly if the sizes of the tab are not the same.
```dart
import 'package:flutter/material.dart';
import 'package:flutter/scheduler.dart';

void main() {
  timeDilation = 10.0;
  runApp(Test());
}

class Test extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: '',
      home: Material(
        color: Colors.black,
        child: DefaultTabController(
          length: 3,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Text('Tap 1 then tap 2.', style: TextStyle(color: Colors.white)),
              TabBar(
                isScrollable: true,
                indicatorSize: TabBarIndicatorSize.tab,
                tabs: <Widget>[
                  Text('-', style: TextStyle(fontSize: 48.0)),
                  Text('1ljkjlkjlk', style: TextStyle(fontSize: 48.0)),
                  Text('2', style: TextStyle(fontSize: 48.0)),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

Reply:
You are right. It doesn't work properly. It seems my current approach is wrong.
Sorry for wasting your time. 🙏